jni-glue/src/vm.rs

```diff
@@ -1,37 +1,43 @@
 use super::*;
 
 /// FFI: Use **&VM** instead of *const JavaVM.  This represents a global, process-wide Java exection environment.
-/// 
+///
 /// On Android, there is only one VM per-process, although on desktop it's possible (if rare) to have multiple VMs
 /// within the same process.  While this library does not yet support having multiple VMs active simultaniously, please
 /// don't hesitate to [file an issue](https://github.com/MaulingMonkey/jni-bindgen/issues/new) if this is an important
 /// use case for you.
 ///
 /// This is a "safe" alternative to jni_sys::JavaVM raw pointers, with the following caveats:
-/// 
+///
 /// 1)  A null vm will result in **undefined behavior**.  Java should not be invoking your native functions with a null
 ///     *mut JavaVM, however, so I don't believe this is a problem in practice unless you've bindgened the C header
 ///     definitions elsewhere, calling them (requiring `unsafe`), and passing null pointers (generally UB for JNI
 ///     functions anyways, so can be seen as a caller soundness issue.)
-/// 
+///
 /// 2)  Allowing the underlying JavaVM to be modified is **undefined behavior**.  I don't believe the JNI libraries
 ///     modify the JavaVM, so as long as you're not accepting a *mut JavaVM elsewhere, using unsafe to dereference it,
 ///     and mucking with the methods on it yourself, I believe this "should" be fine.
 #[repr(transparent)]
-pub struct VM(JavaVM);
+pub struct VM(*const JavaVM);
 impl VM {
-    pub fn as_java_vm(&self) -> *const JavaVM { &self.0 }
-    pub unsafe fn from_jni_local(vm: &JavaVM) -> &VM { &*(vm as *const JavaVM as *const VM) }
+    pub fn as_java_vm(&self) -> *mut JavaVM {
+        self as *const VM as *mut JavaVM
+    }
+    pub unsafe fn from_jni_local(vm: &JavaVM) -> &VM {
+        &*(vm as *const JavaVM as *const VM)
+    }
```

Comment:
Unless I'm missing something major these are both super bogus if `VM` is changed to contain `*const JavaVM`.

The whole `#[repr(transparent)]` and cast dance is so `&VM` / `*VM` are equivalent to `&JavaVM` / `*JavaVM` in ABI layout etc. and allows the use of `&'a VM` to constrain the duration of VM access.  A similar song and dance is performed by `Env`, which is more tested and works fine.  I'm skeptical this is the correct fix.

Reply:
Yeah, I'm not sure it is either. Any thoughts where the extra deref could be happening?